src/Squirrel.CommandLine/SquirrelHost.cs

```diff
@@ -1,111 +1,75 @@
-                    { "s3-up", "Upload releases to S3 API", new SyncS3Options(), o => Upload(new S3Repository(o)) },
-                    { "github-down", "Download latest release from GitHub", new SyncGithubOptions(), o => Download(new GitHubRepository(o)) },
-                    { "github-up", "Upload latest release to GitHub", new SyncGithubOptions(), o => Upload(new GitHubRepository(o)) },
-                };
+            case "mac":
+            case "osx":
+            case "macos":
+                if (!SquirrelRuntimeInfo.IsOSX)
+                    logger.Write("Cross-compiling will cause some features of Squirrel to be disabled.", LogLevel.Warn);
+                packageCommands = OSX.Commands.GetCommands();
+                break;
 
-                if (verbose) {
-                    logger.Level = LogLevel.Debug;
-                }
-
-                if (help) {
-                    commands.WriteHelp();
-                    return 0;
-                }
+            default:
+                throw new NotSupportedException("Unsupported OS platform: " + xplat);
+            }
 
-                try {
-                    // parse cli and run command
-                    commands.Execute(restArgs.ToArray());
-                    return 0;
-                } catch (Exception ex) when (ex is OptionValidationException || ex is OptionException) {
-                    // if the arguments fail to validate, print argument help
-                    Console.WriteLine();
-                    logger.Write(ex.Message, LogLevel.Error);
-                    commands.WriteHelp();
-                    Console.WriteLine();
-                    logger.Write(ex.Message, LogLevel.Error);
-                    return -1;
-                }
-            } catch (Exception ex) {
-                // for other errors, just print the error and short usage instructions
-                Console.WriteLine();
-                logger.Write(ex.ToString(), LogLevel.Error);
-                Console.WriteLine();
-                Console.WriteLine(sqUsage);
-                Console.WriteLine($" > 'csq -h' to see program help.");
-                return -1;
+            if (verbose) {
+                logger.Level = LogLevel.Debug;
             }
-        }
 
-        static void Upload<T>(T repo) where T : IPackageRepository => repo.UploadMissingPackages().GetAwaiter().GetResult();
+            RootCommand rootCommand = new RootCommand($"Squirrel {SquirrelRuntimeInfo.SquirrelDisplayVersion} for creating and distributing Squirrel releases.");
+            rootCommand.AddGlobalOption(PlatformOption);
+            rootCommand.AddGlobalOption(VerboseOption);
+            foreach (var command in packageCommands) {
+                rootCommand.Add(command);
+            }
+            Command deploymentCommand = new("deployment", "Command for deploying Squirrel releases") {
```

Comment:
This is deviation from the original CLI structure as it nests the deployment tasks. I can certainly flatten these back up pretty easily by simply dropping the parent commands and adjusting the command names.

Reply:
I have pushed a commit to your PR which reverses the command nesting. Instead of `csq deployment s3 upload ...` it's now just `csq upload s3 ...`. Do you agree with this change?

Reply:
Yea I like this.